opennebula/resource_opennebula_service_template.go

```diff
@@ -0,0 +1,395 @@
+package opennebula
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+
+	"github.com/OpenNebula/one/src/oca/go/src/goca"
+	srv_tmpl "github.com/OpenNebula/one/src/oca/go/src/goca/schemas/service_template"
+)
+
+func resourceOpennebulaServiceTemplate() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceOpennebulaServiceTemplateCreate,
+		Read:   resourceOpennebulaServiceTemplateRead,
+		Exists: resourceOpennebulaServiceTemplateExists,
+		Update: resourceOpennebulaServiceTemplateUpdate,
+		Delete: resourceOpennebulaServiceTemplateDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Optional:    true,
```

Comment:
It seems this parameter is "Required". Can you use "Required: true" instead

Thanks to this you can remove line 98, Terraform will do the job for you ;)

Reply:
The name is required, but if you pass it in the JSON template there's no need to pass it there too. 

As you mentioned in another comment it would be nice to directly manage the template with GOCA (as with VMs) which probably will help to this case. But that's require some work, if people start using it and asking for that feature probably we can add it in the future. Note that the JSON schema is complex and it's difficult to improve it without introducing breaking changes.

Reply:
OK Understood :)